hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java

```diff
@@ -362,6 +362,14 @@ public FSDataOutputStream create(final Path f,
       final short replication,
       final long blockSize,
       final Progressable progress) throws IOException {
+    return createInternal(f, permission, overwrite, blockSize, false);
+  }
+
+  private FSDataOutputStream createInternal(final Path f,
```

Comment:
Seems like this refraction we are doing only for Blob Endpoint. this new parameter is not used by DFS Client. Can we have these handling only in ABFSBlobClient?

Let's discuss this once offline if its possible.

Reply:
The new field is just to tell BlobClient, if the createPath is for createNonRecursive HDFS API or create HDFS API. The required orchestration for blob on createNonRecursive is done in blobClient only. This field is just to propagate the information to the client about what HDFS API has invoked it.